src/cocoa_joystick.m

```diff
@@ -496,9 +508,9 @@ int _glfwPlatformJoystickPresent(int joy)
 const char* _glfwPlatformGetJoystickName(int joy)
 {
     _GLFWjoydeviceNS* joystick = _glfw.ns_js.devices + joy;
-    if (!pollJoystickEvents(joystick))
+    if (joystick->present)
+        return joystick->name;
+    else
```

Comment:
Please follow the pattern above of checking failure cases and then successful return at the end.